kodi: update to 17.0-beta1-81d5d26

MilhouseVH opened the pull request:
Bump `ffmpeg` to 3.1.2-Krypton-Beta1, `expat` to 2.2.0 and `curl` to 7.5.1 to be in sync with same versions used by Kodi 17-b1.

RPi/RPi2 backports updated - @popcornmix can you eyeball this to make sure I've not left out any "unstable" commits that are now considered stable (I can force push if you need to rebase newclock5).

MilhouseVH commented:
Updated after a few RPi commits were shuffled about (thanks @popcornmix).

Not yet fully build tested, I'll do that today/tomorrow.

MilhouseVH commented:
Apparently Odroid_C2 will have problems building Kodi 17 beta1:
```
      APPLY PATCH (project)   /home/pfau/dev/LibreELEC.tv/projects/Odroid_C2/patches/kodi/kodi-0001-owersun-odroid.patch
patching file configure.ac
Hunk #1 succeeded at 214 (offset 1 line).
Hunk #2 succeeded at 479 (offset 1 line).
Hunk #3 succeeded at 490 (offset 1 line).
Hunk #4 succeeded at 1503 (offset 12 lines).
Hunk #5 succeeded at 1741 (offset 12 lines).
Hunk #6 succeeded at 2132 (offset 3 lines).
Hunk #7 FAILED at 2328.
1 out of 7 hunks FAILED -- saving rejects to file configure.ac.rej
patching file m4/xbmc_arch.m4
patching file xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
patching file xbmc/cores/VideoPlayer/DVDCodecs/DVDFactoryCodec.cpp
patching file xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
Hunk #1 FAILED at 398.
Hunk #2 succeeded at 713 (offset -17 lines).
Hunk #3 succeeded at 1563 (offset -17 lines).
Hunk #4 succeeded at 1628 (offset -17 lines).
Hunk #5 succeeded at 2154 (offset -33 lines).
Hunk #6 succeeded at 2175 (offset -33 lines).
Hunk #7 succeeded at 2247 (offset -33 lines).
1 out of 7 hunks FAILED -- saving rejects to file xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp.rej
The next patch would create the file xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecMFC.cpp,
```
Does anyone have an updated patch that I can add to this PR?

kszaq commented:
I think at this point we can drop the Oversun's Kodi patch and use WeTek Hub Kodi patches. The downside would be not having all C2's resolution available to choose for Kodi but this can be added later.

mapfau commented:
agree, the oversun patch is a mixture for all odroid devices (V4L2 / libHybris)
There are some sound parameters adjusted in Oversun's C2 patch, not sure if wetek HUB needs this also.

kszaq commented:
I don't think these are needed at this point, we still need to PR patches for HD audio passthrough. I haven't had a look at Krypton yet and I tell from my experience with Jarvis.

mapfau commented:
ok, i'll try in the meantime with oversuns patch. Looking forward to test codesnakes work.

MilhouseVH commented:
peripheral.joystick master is no longer compatible with Kodi 17 beta1 - use the last version that is.

MilhouseVH commented:
Build and run-time tested this PR on RPi/RPi2/Generic.

Seems OK, although there is one problem - if the user has the current repository version of `inputstream.mpd` installed (v1.2.0), this will cause kodi.bin to crash on startup: [crashlog](http://sprunge.us/EUCK)

```
Thread 1 (Thread 0x7f761a31e840 (LWP 1251)):
#0  0x00007f76187836f2 in ?? () from /storage/.kodi/addons/inputstream.mpd/inputstream.mpd.so.1.2.0
#1  0x00007f7618783811 in GetStreamIds () from /storage/.kodi/addons/inputstream.mpd/inputstream.mpd.so.1.2.0
#2  0x0000000000948d53 in ADDON::CInputStream::CheckAPIVersion() ()
#3  0x000000000091f9b3 in ADDON::CAddonDll<DllAddon<InputStreamAddonFunctions, INPUTSTREAM_PROPS>, InputStreamAddonFunctions, INPUTSTREAM_PROPS>::Create() ()
#4  0x0000000000949a6c in ADDON::CInputStream::UpdateConfig() ()
#5  0x0000000000949d9e in ADDON::CInputStream::CheckConfig() ()
#6  0x0000000000949e8b in ADDON::CInputStream::FromExtension(ADDON::AddonProps, cp_extension_t const*) ()
#7  0x00000000009653cc in ADDON::CAddonBuilder::Build() ()
#8  0x00000000009380e0 in ADDON::CAddonMgr::GetAddonsInternal(ADDON::TYPE const&, std::vector<std::shared_ptr<ADDON::IAddon>, std::allocator<std::shared_ptr<ADDON::IAddon> > >&, bool) ()
#9  0x0000000000933a7e in ADDON::CBinaryAddonCache::Update() ()
#10 0x0000000000c28451 in CServiceManager::Init2() ()
#11 0x0000000000bd94a5 in CApplication::Create() ()
#12 0x0000000000ab48f8 in XBMC_Run ()
#13 0x00000000007ce7b0 in main ()
```
Which could be a problem for users upgrading from 7.90.004 or any earlier version when `inputstream.mpd` is installed. @mapfau any ideas/thoughts?

mapfau commented:
Maybe there has to be a different version check for this - will look over it over the day

mapfau commented:
I'll build a try/catch patch around it - the situation is kind of ugly because it was forgotten to implement the API version

MilhouseVH commented:
Thanks - ugly is better than kodi not being able to start. If you able to come up with something we can add it to this PR.

MilhouseVH commented:
Add `libhdhomerun` dependency otherwise `pvr.hdhomerun` won't work... could install it in the add-on but meh.

escalade commented:
Peripheral.joystick seems to be broken. Controllers show up as connected but buttons aren't registered when trying to map the buttons. Tested with DS4.

mapfau commented:
Odroid_C2 video is not running at all because of missing AML video /sys/parameter/.... value which is be used to detect if its kernel 3.14.
This parameter exist on WETEK HUB, but not on Odroid C2, so there are 2 choices to fix:

1.) Make a kodi patch, wich does not check for this value
2.) Make a kernel Patch wich writes this module parameter.

I'm not LibreElec professional and  therefore unsure how to solve such issues.
But I can point to the place wich has to be solved. Can one of you pls. let me know how to fix?

I'm currently using a simple patch for Odroid_C2/kodi patch.

escalade commented:
Joystick.peripherals is broken with beta1, but there's a fix: 

https://github.com/garbear/xbmc/commit/ee2dc577a03d86ac61e06bc95aaff1d1eddbf593

Also, I used to be able to pair from my phone (with bluetooth after running hciconfig hci0 piscan). A passkey confirmation would pop up in Kodi. This no longer happens with beta1, although I can pair through the LibreELEC bluetooth menus fine. I'm guessing the functionality for this is in LibreELEC-settings? Should I report an issue there? I downgraded to a build with the previous Kodi version and it works fine again.

MilhouseVH commented:
@escalade: I've added https://github.com/garbear/xbmc/commit/ee2dc577a03d86ac61e06bc95aaff1d1eddbf593 to this PR, can you confirm it is OK?

Odroid_C2: I've added an updated oversun commit from [@mapfau](https://github.com/mapfau/LibreELEC.tv/blob/6be9637573cd3ebe3e5b4ce721d8af1e64df0a6f/projects/Odroid_C2/patches/kodi/kodi-0001-owersun-odroid.patch), in his own words: "its cleaned so all odroid things not part of odroidc2 are in there". @mapfau please test & thanks!

escalade commented:
Yes, it's tested and fixes the issue.

On Friday, 26 August 2016, MilhouseVH <notifications@github.com> wrote:

> @escalade <https://github.com/escalade>: I've added garbear/xbmc@ee2dc57
> <https://github.com/garbear/xbmc/commit/ee2dc577a03d86ac61e06bc95aaff1d1eddbf593>
> to this PR, can you confirm it is OK?
>
> Odroid_C2: I've added an updated oversun commit from @mapfau
> <https://github.com/mapfau/LibreELEC.tv/blob/6be9637573cd3ebe3e5b4ce721d8af1e64df0a6f/projects/Odroid_C2/patches/kodi/kodi-0001-owersun-odroid.patch>,
> in his own words: "its cleaned so all odroid things not part of odroidc2
> are in there". @mapfau <https://github.com/mapfau> please test & thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LibreELEC/LibreELEC.tv/pull/641#issuecomment-242834615>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB7rdTou5r5Ees3cy4lEkoEJ-y1feFS6ks5qj0QLgaJpZM4JpaAO>
> .
>


-- 

Sent from Gmail

MilhouseVH commented:
I know but please test this PR (just to be sure I haven't cocked anything up!)

chewitt commented:
@MilhouseVH thanks for your efforts on this, there were lots of kinks to work out :)
